static/src/javascripts/projects/commercial/modules/messenger.ts

```diff
@@ -0,0 +1,382 @@
+import reportError from '../../../lib/report-error';
+import { postMessage } from './messenger/post-message';
+
+/**
+ * A message that is sent from an iframe following a standard format
+ *
+ * TODO Is this format formally defined somewhere?
+ */
+type StandardMessage<T = unknown> = {
+	id: string;
+	type: string;
```

Comment:
Looking into this myself, I wondered about pre-populating `LISTENERS` with all the possible message types.

```
type MessageType = 'background' | 'click' | 'get-page-url' | 'get-styles' | 'measure-ad-load' | 'resize' | 'set-ad-height' | 'type';
const LISTENERS: Record<MessageType, undefined | Array<() => void>> = {
        'get-page-url': undefined,
        'get-styles': undefined,
        'measure-ad-load': undefined,
        'set-ad-height': undefined,
        background: undefined,
        click: undefined,
        resize: undefined,
        type: undefined,
};
```

Reply:
This is a really interesting idea, will take a look at implementing this!

Reply:
Added something like this!